src/category_theory/types.lean

```diff
@@ -16,14 +38,35 @@ instance types : large_category (Type u) :=
   id      := λ a, id,
   comp    := λ _ _ _ f g, g ∘ f }
 
-@[simp] lemma types_hom {α β : Type u} : (α ⟶ β) = (α → β) := rfl
-@[simp] lemma types_id (X : Type u) : 𝟙 X = id := rfl
-@[simp] lemma types_comp {X Y Z : Type u} (f : X ⟶ Y) (g : Y ⟶ Z) : f ≫ g = g ∘ f := rfl
+lemma types_hom {α β : Type u} : (α ⟶ β) = (α → β) := rfl
+@[simp, priority 10] lemma types_id (X : Type u) : 𝟙 X = id := rfl
+@[simp, priority 10] lemma types_comp {X Y Z : Type u} (f : X ⟶ Y) (g : Y ⟶ Z) : f ≫ g = g ∘ f := rfl
```

Comment:
These two simp lemmas also seem dangerous to me.  I would imagine they break `functor.map_id` and `functor.map_comp`.

And indeed it breaks:
```lean
example (F : Type ⥤ Type) (f g : ℕ → ℕ) : F.map (f ≫ g) = F.map f ≫ F.map g :=
by simp -- fails
```

Reply:
Also fails with the harpoons:
```lean
example (F : Type ⥤ Type) (f g : ℕ → ℕ) : F.map (↾f ≫ ↾g) = F.map ↾f ≫ F.map ↾g :=
by simp
```

Reply:
I'd hoped that merely lowering their priority was going to let me keep them, but I'll investigate dropping them.

Reply:
I've removed `@[simp]` on these, and added lemmas for these facts applied to terms.

The only place I had to change otherwise was actually outside the category theory library, where `topology/metric_space/lipschitz.lean` needed `types_id` added back into a simp set!